lib/pages/dashboard/dashoard_page.dart

```diff
@@ -22,17 +38,53 @@ class _DashboardPageState extends State<DashboardPage> {
       sliver: SliverList(
         delegate: SliverChildListDelegate.fixed([
           Paragraph(title: "Vos clubs"),
-          DashboardClubs(onClubChange: (club) => _selectCurrentClub(club)),
+          DashboardClubs(key: ValueKey("dashboard-clubs"), onClubChange: (club) => _selectCurrentClub(club)),
           Paragraph(title: "Vos équipes"),
-          if (_currentClub != null) DashboardClubTeams(club: _currentClub, onTeamFavoriteChange: (_) => _selectCurrentClub(_currentClub)),
+          if (_currentClub != null) _buildDashboardElement(_teamsElement),
           Paragraph(title: "Votre agenda"),
-          DashboardAgenda()
+          if (_currentClub != null) _buildDashboardElement(_agendaElement)
         ])
       )
     );
   }
 
+  Widget _teamsElement(List<Team> teams) {
+    if (teams.length > 0) {
+      return DashboardClubTeams(teams: teams, onFavoriteTeamsChange: () => _selectCurrentClub(_currentClub));
+    } else {
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(28.0),
+          child: RaisedButton(
+            onPressed: () =>  Router.push(context: context, builder: (_) => ClubDetailPage(_currentClub)).then(
+              (_) => _selectCurrentClub(_currentClub),
+            ),
+            padding: EdgeInsets.all(12.0),
+            child: Text("Sélectionnez une équipe favorite")
+          ),
+        ),
+      );
+    }
+  }
+
+  Widget _agendaElement(List<Team> teams) {
```

Comment:
Is it easier to understand, instead of using `DashboardAgenda(teams: teams)` at line https://github.com/eric-taix/volley34-mobile/pull/79/files#diff-30d704deb5d63a4c0851bea4681f954cR45 ?

Reply:
I used that to encapsulate the agenda in a `BlocBuilder` widget.